assets/src/edit-story/components/panels/textStyle/font.js

```diff
@@ -47,16 +47,53 @@ function FontControls({ selectedElements, pushUpdate }) {
   const fontFamily = getCommonValue(selectedElements, 'fontFamily');
   const fontSize = getCommonValue(selectedElements, 'fontSize');
   const fontWeight = getCommonValue(selectedElements, 'fontWeight');
+  const timeout = useRef(null);
 
   const {
     state: { fonts },
-    actions: { getFontWeight, getFontFallback },
+    actions: { maybeEnqueueFontStyle, getFontWeight, getFontFallback },
   } = useFont();
   const fontWeights = useMemo(() => getFontWeight(fontFamily), [
     getFontWeight,
     fontFamily,
   ]);
 
+  const loadFont = useCallback(
+    (family, callback) => {
+      // 0. Check browser support to load fonts using JavaScript
+      if (document?.fonts) {
+        const font = `0 ${family}`;
+
+        // 1. Check if font family was not loaded yet
+        if (!document.fonts.check(font)) {
+          // 2. Try to inject <link /> resource of the font family
+          maybeEnqueueFontStyle(family);
```

Comment:
That should always be called before anything else. `fonts.check` is not a good `if` for this.